install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml

```diff
@@ -445,7 +445,6 @@ spec:
       priorityClassName: {{ include "cilium.priorityClass" (list $ .Values.priorityClassName "system-node-critical") }}
       serviceAccount: {{ .Values.serviceAccounts.cilium.name | quote }}
       serviceAccountName: {{ .Values.serviceAccounts.cilium.name | quote }}
-      terminationGracePeriodSeconds: 1
```

Comment:
just curious why this can't be increased to something bigger (e.g. 60s) so that TERM signal will be sent.

Reply:
The reason for it to be `1` is to terminate the cilium-agent quickly so that the new agent restarts immediately. This should be kept as it is.

Reply:
But  if in `1` seconds, `cilium-agent` does not finish it tasks, it'll be **force killed**, and can lead to unexpected state.
@aanm Do you think we should put correctness first.  
In k8s there are many ways to speed up rolling update other than `terminationGracePeriodSeconds`, e.g inc `rollingUpdate.maxSurge`

Reply:
>it'll be force killed, and can lead to unexpected state.

Yes, that is intended as Cilium can recover its state on startup. We can discuss the removal of this option in a different PR.

Reply:
Yeah, I undo this change to discuss in other PR.
@aanm Could you take a final look?